script/deployments/verifyDeployment.sh

```diff
@@ -0,0 +1,136 @@
+
+# Check if deployment directory is provided
+if [ "$#" -ne 1 ]; then
+    echo "Usage: $0 <deployment_directory>"
+    exit 1
+fi
+
+# Get the script's directory
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+
+# Set the project root directory (two levels up from the script)
+PROJECT_ROOT="$( cd "$SCRIPT_DIR/../.." &> /dev/null && pwd )"
+
+# Set the deployment directory from the command line argument
+DEPLOYMENT_DIR="$1"
+
+# Ensure DEPLOYMENT_DIR is an absolute path
+if [[ "$DEPLOYMENT_DIR" != /* ]]; then
+    DEPLOYMENT_DIR="$SCRIPT_DIR/$DEPLOYMENT_DIR"
+fi
+
+# Set the environment variables
+source "$PROJECT_ROOT/.env"
+
+# Get optimizer runs from forge config
+OPTIMIZER_RUNS=$(forge config --json | jq -r '.optimizer_runs')
+
+# Find the log file
+LOG_FILE=$(find "$DEPLOYMENT_DIR" -name "gnosischain-*.log" | head -n 1)
+
+# Compiler version set in foundry.toml
+COMPILER_VERSION="v0.8.24+commit.e11b9ed9"
+
+echo "Using Compiler Version: $COMPILER_VERSION"
+echo "Using Optimizer Runs: $OPTIMIZER_RUNS"
+
+# Function to ABI-encode constructor arguments
+abi_encode_constructor_args() {
+    local contract_name=$1
+    local constructor_args_file=$2
+    local abi_file="${PROJECT_ROOT}/out/${contract_name}.sol/${contract_name}.json"
+    
+    # Extract constructor types from ABI
+    local constructor_types=$(jq -r '.abi[] | select(.type == "constructor") | .inputs | map(.type) | join(",")' "$abi_file")
+    
+    if [ -z "$constructor_types" ]; then
+        echo "No constructor found or empty constructor" >&2
+        return
+    fi
+    
+    echo "Constructor types: $constructor_types" >&2
+    
+    # Read constructor arguments (single line, space-separated)
+    local constructor_args=$(cat "$constructor_args_file")
+    echo "Constructor arguments: $constructor_args" >&2
+    
+    # ABI-encode the arguments
+    local encoded_args=$(cast abi-encode "constructor(${constructor_types})" $constructor_args)
+    
+    if [ $? -ne 0 ]; then
+        echo "Error encoding constructor arguments" >&2
+        return 1
+    fi
+    
+    echo "${encoded_args:2}"  # Remove '0x' prefix
+}
+
+
+# Function to verify a contract
+verify_contract() {
+    local contract_name=$1
+    local deployed_address=$2
+    local source_path=$3
+    local constructor_args_file=$4
+
+    echo "Verifying ${contract_name}..."
+
+    # ABI-encode constructor arguments
+    local encoded_args=$(abi_encode_constructor_args "$contract_name" "$constructor_args_file")
+    
+    if [ $? -ne 0 ]; then
+        echo "Failed to encode constructor arguments. Skipping verification for ${contract_name}."
+        return 1
+    fi
+
+    echo "ABI-encoded constructor arguments: $encoded_args"
+
+    # Attempt verification
+    if forge verify-contract \
+        --chain-id 100 \
+        --compiler-version ${COMPILER_VERSION} \
+        --optimizer-runs ${OPTIMIZER_RUNS} \
+        --flatten \
+        --force \
+        --evm-version "cancun" \
```

Comment:
for some reason gnosisscan just does not want to use "cancun" over the API, despite it being specified here